Fix star imports f2py

charris opened the pull request:
Also bite the bullet and make PEP8 fixes in `numpy/f2py/*.py`. That cleans up the code remarkably, leaving
it less incomprehensible. Some errors and potential errors were also detected in fixing the `import *` imports.

charris committed:
STY: Make pyflakes fixes in numpy/f2py

charris committed:
STY: Make PEP8 fixes in numpy/f2py

Decided to bite the bullet on this one. The code is certainly more
readable, so should be easier to fix if we need to.

charris committed:
STY: PEP8 fixes in numpy/f2py/tests.

jaimefrio commented on numpy/f2py/auxfuncs.py:
```diff
@@ -40,44 +65,57 @@ def outmess(t):
+
 def _isstring(var):
-    return 'typespec' in var and var['typespec']=='character' and (not isexternal(var))
+    return 'typespec' in var and var['typespec'] == 'character' and (not isexternal(var))
```
This fits in 79 columns?

charris replied:
No:) I ignored long lines for this PR, some are well over a 100 characters.

jaimefrio commented on numpy/f2py/auxfuncs.py:
```diff
@@ -88,187 +126,221 @@ def get_kind(var):
+
 def ismodule(rout):
-    return ('block' in rout and 'module'==rout['block'])
+    return ('block' in rout and 'module' == rout['block'])
```
No parenthesis needed around the return.

jaimefrio commented on numpy/f2py/auxfuncs.py:
```diff
@@ -88,187 +126,221 @@ def get_kind(var):
 
 def isfunction(rout):
-    return ('block' in rout and 'function'==rout['block'])
+    return ('block' in rout and 'function' == rout['block'])
```
Extra () here again

jaimefrio commented on numpy/f2py/auxfuncs.py:
```diff
@@ -88,187 +126,221 @@ def get_kind(var):
-    return ('block' in rout and 'function'==rout['block'])
+    return ('block' in rout and 'function' == rout['block'])
+
+# def isfunction_wrap(rout):
```
Isn't the policy to remove commented out code?

charris replied:
Yes, but with this code I'm never sure if commented out code might be informative as to intent.

jaimefrio commented on numpy/f2py/auxfuncs.py:
```diff
@@ -88,187 +126,221 @@ def get_kind(var):
+
 def issubroutine(rout):
-    return ('block' in rout and 'subroutine'==rout['block'])
+    return ('block' in rout and 'subroutine' == rout['block'])
```
Extra ()

rgommers commented:
New, now only 88.9% incomprehensible:)

More seriously, thanks for taking on all this housekeeping Chuck.

charris committed:
STY: Break some long lines in numpy/f2py/*.py.

The fixes are generated by autopep8, which uses line continuation.
There are 441 cases that it is unable to handle, involving strings, and
that is more, and more delicate, work than I want to do at this time.
The line continuation characters at least mark some of the long lines.

charris commented:
Still many long lines, mostly strings. Breaking those is a lot of somewhat delicate work and more than I want to do at this time.
